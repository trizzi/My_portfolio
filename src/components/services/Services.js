import React from 'react';
import services from './services.json';
import ServiceItem from './ServiceItem';
import serviceImage from '../../assets/services_background_image_lap.svg';
import serviceLogo from '../../assets/Button (3).svg';

const Services = () => {
	return (
		<div
			name='section4'
			className='container-xl mx-auto py-10 lg:py-32'
			style={{
				backgroundImage: `url(${serviceImage})`,
				backgroundRepeat: 'no-repeat',
				backgroundSize: 'cover',
				backgroundColor: 'rgba(0,0,0,0.9)',
			}}>
			<div>
				<div className='flex justify-center'>
					<div>
						<img
							src={serviceLogo}
							alt='service-logo'
							className='mx-auto h-24 rounded-full sm:mx-0 sm:shrink-0'
						/>
					</div>

					<h1 className='text-4xl text-bold pb-6'>Services</h1>
				</div>
			</div>
			<div className='container mx-auto p-8 md:p-20'>
				<div className='relative flex flex-wrap w-full lg:justify-center lg:space-x-6 '>
					{services.map((service) => (
						<ServiceItem
							key={service.id}
							service={service}
						/>
					))}
				</div>
			</div>
		</div>
	);
};

export default Services;
