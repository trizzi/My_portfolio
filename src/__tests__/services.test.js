import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Services from '../components/services/Services';

import { ThemeContextProvider } from '../context/ThemeContext';

// Mock the services data
jest.mock('../components/services/services.json', () => [
	{
		id: 1,
		heading: 'Service 1',
		text: 'Description 1',
		name: 'Name 1',
		image: 'Image 1',
	},
	{
		id: 2,
		heading: 'Service 2',
		text: 'Description 2',
		name: 'Name 2',
		image: 'Image 2',
	},
	{
		id: 3,
		heading: 'Service 3',
		text: 'Description 3',
		name: 'Name 3',
		image: 'Image 3',
	},
]);

test('renders Services section with correct content', () => {
	const { container } = render(
		<ThemeContextProvider>
			<Services />
		</ThemeContextProvider>,
	);

	// Check if the services section is rendered
	const servicesSection = screen.getByTestId('service-section');
	expect(servicesSection).toBeInTheDocument();

	// Get the number of services rendered
	const services = screen.getAllByRole('heading', { level: 3 });
	const numServices = services.length;

	// Check if the service items are rendered with the correct data
	const serviceTitles = screen.getAllByRole('heading', { level: 3 }); // Assuming service titles are rendered as h3 elements
	expect(serviceTitles).toHaveLength(numServices); // Assuming there are three services

	const serviceDescriptions = screen.getAllByText(/Description/i); // Matches any text containing "Description"
	expect(serviceDescriptions).toHaveLength(numServices); // Assuming there are three services

	expect(container).toMatchSnapshot();
});
