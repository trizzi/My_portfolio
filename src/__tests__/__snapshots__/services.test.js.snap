// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Services section with correct content 1`] = `
<div>
  <div
    class="container-xl mx-auto py-10 lg:py-32"
    data-testid="service-section"
    name="section4"
    style="background-image: url(services_background_image_lap.svg); background-repeat: no-repeat; background-size: cover; background-color: rgba(0, 0, 0, 0.9);"
  >
    <div>
      <div
        class="flex justify-center"
      >
        <div>
          <img
            alt="service-logo"
            class="mx-auto h-24 rounded-full sm:mx-0 sm:shrink-0"
            src="Button-3.svg"
          />
        </div>
        <h1
          class="text-4xl text-bold pb-6"
        >
          Services
        </h1>
      </div>
    </div>
    <div
      class="container mx-auto p-8 md:p-20"
    >
      <div
        class="relative flex flex-wrap w-full lg:justify-center lg:space-x-6 "
      >
        <div
          class="w-full  rounded-xl overflow-hidden shadow-xl p-5 xl:px-10  mt-10 lg:mt-0 mr-6 lg:mr-0 md:w-2/5 lg:w-1/4  bg-[#292A3C] opacity-80"
        >
          <div
            class="flex justify-between"
          >
            <h1
              class="font-bold text-xl"
            >
              Service 1
            </h1>
            <img
              alt="service_image"
              class=" w-[64px] h-[64px]"
              src="Image 1"
            />
          </div>
          <div
            class="flex-grow border-t border-[#9932CC] shadow-xl"
          />
          <h3
            class="py-5"
          >
            Name 1
          </h3>
          <p>
            Description 1
          </p>
        </div>
        <div
          class="w-full  rounded-xl overflow-hidden shadow-xl p-5 xl:px-10  mt-10 lg:mt-0 mr-6 lg:mr-0 md:w-2/5 lg:w-1/4  bg-[#292A3C] opacity-80"
        >
          <div
            class="flex justify-between"
          >
            <h1
              class="font-bold text-xl"
            >
              Service 2
            </h1>
            <img
              alt="service_image"
              class=" w-[64px] h-[64px]"
              src="Image 2"
            />
          </div>
          <div
            class="flex-grow border-t border-[#9932CC] shadow-xl"
          />
          <h3
            class="py-5"
          >
            Name 2
          </h3>
          <p>
            Description 2
          </p>
        </div>
        <div
          class="w-full  rounded-xl overflow-hidden shadow-xl p-5 xl:px-10  mt-10 lg:mt-0 mr-6 lg:mr-0 md:w-2/5 lg:w-1/4  bg-[#292A3C] opacity-80"
        >
          <div
            class="flex justify-between"
          >
            <h1
              class="font-bold text-xl"
            >
              Service 3
            </h1>
            <img
              alt="service_image"
              class=" w-[64px] h-[64px]"
              src="Image 3"
            />
          </div>
          <div
            class="flex-grow border-t border-[#9932CC] shadow-xl"
          />
          <h3
            class="py-5"
          >
            Name 3
          </h3>
          <p>
            Description 3
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
